### ISSUE TITLE
Error: Debug Failure. False expression: Changes overlap

### ISSUE THREAD

**Issue by oblitum** [neoclide/coc-tsserver#212]
Hi, I rarely code js, but just started some session and got a very long trace due to an error that seems fixed upstream:

- https://github.com/microsoft/TypeScript/issues/27700

In coc it happens when querying for `"quickfixes"`. So is https://github.com/microsoft/TypeScript/pull/28217 merged in this fork?

**Comment by oblitum** [comment]
Hmmm, sorry, the source structure between the two projects seems completely different, even though I get same error.

**Comment by oblitum** [comment]
This is the full log when I hover over something with a quickfix and query for it:

```
Error on request (quickfixes): Vim(return):Error invoking 'quickfixes' on channel 3 (coc):
Error processing request. Debug Failure. False expression: Changes overlap
Verbose Debug Information: {"pos":123,"end":1911} and {"pos":398,"end":405}
Error: Debug Failure. False expression: Changes overlap
Verbose Debug Information: {"pos":123,"end":1911} and {"pos":398,"end":405}
    at _loop_10 (/home/francisco/.config/coc/extensions/node_modules/coc-tsserver/node_modules/typescript/lib/tsserver.js:130016:34)
    at /home/francisco/.config/coc/extensions/node_modules/coc-tsserver/node_modules/typescript/lib/tsserver.js:130022:25
    at Object.mapDefined (/home/francisco/.config/coc/extensions/node_modules/coc-tsserver/node_modules/typescript/lib/tsserver.js:610:30)
    at Object.getTextChangesFromChanges (/home/francisco/.config/coc/extensions/node_modules/coc-tsserver/node_modules/typescript/lib/tsserver.js:130010:27)
    at ChangeTracker.getChanges (/home/francisco/.config/coc/extensions/node_modules/coc-tsserver/node_modules/typescript/lib/tsserver.js:129978:45)
    at Function.ChangeTracker.with (/home/francisco/.config/coc/extensions/node_modules/coc-tsserver/node_modules/typescript/lib/tsserver.js:129426:32)
    at Object.getCodeActions (/home/francisco/.config/coc/extensions/node_modules/coc-tsserver/node_modules/typescript/lib/tsserver.js:132160:64)
    at /home/francisco/.config/coc/extensions/node_modules/coc-tsserver/node_modules/typescript/lib/tsserver.js:130599:77
    at Object.flatMap (/home/francisco/.config/coc/extensions/node_modules/coc-tsserver/node_modules/typescript/lib/tsserver.js:513:25)
    at Object.getFixes (/home/francisco/.config/coc/extensions/node_modules/coc-tsserver/node_modules/typescript/lib/tsserver.js:130599:23)
    at /home/francisco/.config/coc/extensions/node_modules/coc-tsserver/node_modules/typescript/lib/tsserver.js:143796:35
    at Object.flatMap (/home/francisco/.config/coc/extensions/node_modules/coc-tsserver/node_modules/typescript/lib/tsserver.js:513:25)
    at Object.getCodeFixesAtPosition (/home/francisco/.config/coc/extensions/node_modules/coc-tsserver/node_modules/typescript/lib/tsserver.js:143794:23)
    at Proxy.<anonymous> (/home/francisco/.config/coc/extensions/node_modules/coc-styled-components/node_modules/typescript-template-language-service-decorator/lib/template-language-service-decorator.js:144:20)
    at IOSession.Session.getCodeFixes (/home/francisco/.config/coc/extensions/node_modules/coc-tsserver/node_modules/typescript/lib/tsserver.js:154662:64)
    at Session.handlers.ts.Map.ts.getEntries._a.<computed> (/home/francisco/.config/coc/extensions/node_modules/coc-tsserver/node_modules/typescript/lib/tsserver.js:153339:61)
    at /home/francisco/.config/coc/extensions/node_modules/coc-tsserver/node_modules/typescript/lib/tsserver.js:154962:88
    at IOSession.Session.executeWithRequestId (/home/francisco/.config/coc/extensions/node_modules/coc-tsserver/node_modules/typescript/lib/tsserver.js:154953:28)
    at IOSession.Session.executeCommand (/home/francisco/.config/coc/extensions/node_modules/coc-tsserver/node_modules/typescript/lib/tsserver.js:154962:33)
    at IOSession.Session.onMessage (/home/francisco/.config/coc/extensions/node_modules/coc-tsserver/node_modules/typescript/lib/tsserver.js:154986:35)
    at Interface.<anonymous> (/home/francisco/.config/coc/extensions/node_modules/coc-tsserver/node_modules/typescript/lib/tsserver.js:157202:27)
    at Interface.emit (events.js:314:20)
    at Interface._onLine (readline.js:337:10)
    at Interface._normalWrite (readline.js:482:12)
    at Socket.ondata (readline.js:194:10)
    at Socket.emit (events.js:314:20)
    at addChunk (_stream_readable.js:307:12)
    at readableAddChunk (_stream_readable.js:282:9)
    at Socket.Readable.push (_stream_readable.js:221:10)
    at Pipe.onStreamRead (internal/stream_base_commons.js:188:23)
```

**Comment by chemzqm** [comment]
Upgrade your local typescript module.

**Comment by oblitum** [comment]
Ah okay, the problem is in the typescript module... well, I just updated to latest nightly and still same issue :-/

**Comment by oblitum** [comment]
I have reduced the code snippet that causes it and reported here: https://github.com/microsoft/TypeScript/issues/40983. Hopefully they can help.

**Comment by oblitum** [comment]
They opened a PR to fix this, just tested, it works.

**Comment by oblitum** [comment]
Fixed on 4.1.1 (beta version), available in nightly builds.
